src/index.tsx

```diff
@@ -120,11 +118,40 @@ function App() {
 
   const [childLayerIds] = useModelState<string[]>("layers");
 
+  const [basemapState, setBasemapState] = useState<MaplibreBasemapModel | null>(
+    null,
+  );
+
   // Fake state just to get react to re-render when a model callback is called
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [stateCounter, setStateCounter] = useState<Date>(new Date());
   const updateStateCallback = () => setStateCounter(new Date());
 
+  useEffect(() => {
```

Comment:
todo: in the future, we should figure out a way to refactor some of this code out of the top-level `App`, because it's pretty ugly having all of this layer model, basemap model, view model, etc, state in the top-level body.